src/global_paths.rs

```diff
@@ -54,18 +54,23 @@ fn get_default_juliaup_home_path() -> Result<PathBuf> {
 }
 
 pub fn get_paths() -> Result<GlobalPaths> {
+    use anyhow::Context;
+
     let juliauphome = get_juliaup_home_path()?;
 
-    #[cfg(feature = "selfupdate")]
     let my_own_path = std::env::current_exe()
-        .with_context(|| "Could not determine the path of the running exe.")?;
+        .with_context(|| "std::env::current_exe() did not find its own path.")?
+        .canonicalize()
+        .with_context(|| "Failed to canonicalize the path to the Julia launcher.")?;
 
-    #[cfg(feature = "selfupdate")]
-    let juliaupselfbin = my_own_path
+    let juliaupselfexecfolder = my_own_path
         .parent()
         .ok_or_else(|| anyhow!("Could not determine parent."))?
         .to_path_buf();
 
+    let juliaupselfexec = juliaupselfexecfolder
+        .join(format!("juliaup{}", std::env::consts::EXE_SUFFIX));
```

Comment:
So I don't really understand why we are doing this, i.e. why we start with the path to the binary, then take the parent, and then reconstruct the path to the binary here. BUT, that is how the code was before, and I'm nervous about changing it... And this might have been related to deal with some symbolic links stuff that I don't fully understand?

Reply:
Honestly I struggle to see the point of it as well. The only scenario I can think of for the reason of that, is that in windows `current_exe` doesn't return (maybe it didn't in the past?) the `.exe` suffix.
Dealing with symbolic links should be handled by `.canonicalize()` now anyway.